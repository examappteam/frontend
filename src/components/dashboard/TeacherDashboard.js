import React, {Component} from "react"

import ScrollableListMenu from "../common/ScrollableListMenu"
import WideListButtonView from "../common/WideListButtonView"
import LgModal from "../common/modals/LgModal"
import { BrowserRouter as Router, Route, Link} from "react-router-dom";

import "./styles/DashboardStyle.css"
import CreateNewClassDialog from "./dialogs/CreateNewClassDialog";

class TeacherDashboard extends Component {
    constructor() {
        super()
        this.state = {
            categories:[
                [
                    { id: 1, name: "TVT17SPO" },
                    { id: 2, name: "TVT17SPL" },
                    { id: 3, name: "EX16SPI" },
                    { id: 4, name: "EX16SPO" },
                    { id: 5, name: "TVT18SPO" },
                    { id: 6, name: "TVT18SPL" }
                ],
                [
                    { id: 1, name: "Basics of Java", linkedClass: "TVT17SPO" },
                    { id: 2, name: "Basic beginner course", linkedClass: "TVT18SPO" },
                    { id: 3, name: "Object oriented analysis", linkedClass: "TVT18SPL" },
                    { id: 4, name: "Advanced maths", linkedClass: "TVT16SPO" },
                    { id: 5, name: "Best course ever", linkedClass: "TVT15SPO" },
                    { id: 6, name: "International exam", linkedClass: "TVT14SPL" }
                ],
                [
                    { id: 1, name: "Professional English",
                      description: "The Professional English test consists of 30 questions. There’s no time limit, so take your time. You will need headphones or speakers for the listening section. You will get your results as soon as you’ve finished the test.",
                      date: "5.2.2019"},

                    { id: 2,description: "The Unprofessional English test consists of 30 questions. There’s no time limit, so take your time. You will need headphones or speakers for the listening section. You will get your results as soon as you’ve finished the test.", name: "Unprofessional English", date: "14.3.2019" },
                    { id: 3,description: "The Professional Swedish test consists of 30 questions. There’s no time limit, so take your time. You will need headphones or speakers for the listening section. You will get your results as soon as you’ve finished the test.", name: "Professional Swedish", date: "14.4.2019" },
                    { id: 4,description: "The Engineering mathematics test consists of 30 questions. There’s no time limit, so take your time. You will need headphones or speakers for the listening section. You will get your results as soon as you’ve finished the test.", name: "Engineering mathematics", date: "15.3.2019" }
                ]
            ],
            selectedCategoryId: 0,
            selectionId: 0,
            showState: false
        }

        this.onScrollableListItemClicked = this.onScrollableListItemClicked.bind(this)
        this.changeShowState = this.changeShowState.bind(this)
    }
    onScrollableListItemClicked = (category, id) => (e) => {
        console.log("event",e)
        e.preventDefault()
        console.log("ran itemclick", category, id)
            this.setState(() =>{
                return{
                    selectedCategoryId: category,
                    selectionId: id-1                   
                }
                //console.log("states ", this.state.selectedCategoryId, this.state.selectionId)
            })
    }

    changeShowState=()=>{
        this.setState(prevState=>({
                showState: !prevState.showState       
        }));
        console.log("showstate",this.state.showState)
    }

    render() {
        console.log(this.state.categories[this.state.selectedCategoryId][this.state.selectionId])

        return(
            <div>
                
                <div className="pure-g">
                    <div className="pure-u-1-3">
                        <div className="padded-box">
                            <ScrollableListMenu 
                                menuHeader="My classes" 
                                menuItems={this.state.categories[0]}
                                selectedItem={this.state.selectionId}
                                selectedCategory={this.state.selectedCategoryId}
                                category = {0}
                                handler = {this.onScrollableListItemClicked.bind(this)}/>
                                <button onClick={this.changeShowState} className="pure-button pure-button-primary">Create new class</button>
                                <LgModal close={this.changeShowState} show={this.state.showState}>
                                    <CreateNewClassDialog close={this.changeShowState}/>
                                </LgModal>
                                <button className="pure-button pure-button-disabled">Delete selected</button> 
                        </div>                  
                    </div>
                    <div className="pure-u-1-3">
                        <div className="padded-box">
                            <ScrollableListMenu 
                                menuHeader="Ready for evaluation" 
                                menuItems={this.state.categories[1]}
                                selectedItem={this.state.selectionId}
                                selectedCategory={this.state.selectedCategoryId}
                                category = {1}
                                handler = {this.onScrollableListItemClicked.bind(this)}/>
                                <button className="pure-button pure-button-disabled">Evaluate selected</button>
                        </div>  
                    </div>
                    <div className="pure-u-1-3">
                        <div className="padded-box">
                            <ScrollableListMenu 
                                menuHeader="My own exams" 
                                menuItems={this.state.categories[2]}
                                selectedItem={this.state.selectionId}
                                selectedCategory={this.state.selectedCategoryId}
                                category = {2}
                                handler = {this.onScrollableListItemClicked.bind(this)}/>
                                <Link to="/create_exam"><button className="pure-button pure-button-primary">Create new exam</button></Link>
                                
                                <button className="pure-button pure-button-disabled">Delete selected</button>                         
                        </div>  
                    </div>
                    
                </div>
                <div className="pure-g">
                <div className="pure-u-3-24"></div>
                <div className="pure-u-18-24">
                    <div className="padded-box">                   
                        <WideListButtonView title={this.state.categories[this.state.selectedCategoryId][this.state.selectionId].name} exam={this.state.categories[this.state.selectedCategoryId][this.state.selectionId]}/>                     
                    </div>
                </div>
                <div className="pure-u-3-24"></div>
                </div>
            </div>
        )
    }
}

export default TeacherDashboard