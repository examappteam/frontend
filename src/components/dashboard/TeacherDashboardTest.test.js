import React from "react";
import { create } from "react-test-renderer";
import TeacherDashboard from "./TeacherDashboard";

describe("TeacherDashboard", () => {
  test("it matches the snapshot", () => {
    const component = create(<TeacherDashboard />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("TeacherDashboard enable delete exam button after clicking exam", () => {
  test("the delete button becomes enabled when an exam is clicked", () => {
    
    const componentDashboard = create(<TeacherDashboard/>);
    const rootInstanceDashboard = componentDashboard.root;
    const teacherDashboard = rootInstanceDashboard.findByType(React.Component);
    console.log(teacherDashboard.state.selectedCategoryId);
    
    // const deleteExamButton = rootInstanceDashboard.findByProps()
    // // teacherDashboard.getElementById("delete-exam-button");
    // expect(deleteExamButton.props.children).toBe("PROCEED TO CHECKOUT");

    const component = create(<Button text="SUBSCRIBE TO BASIC" />);
    const rootInstance = component.root;
    const button = rootInstance.findByType("button");
    button.props.onClick();
    expect(button.props.children).toBe("PROCEED TO CHECKOUT");
  });
});

import Button from "./Button";
describe("Button component", () => {
  test("it shows the expected text when clicked", () => {
    const component = create(<Button text="SUBSCRIBE TO BASIC" />);
    const rootInstance = component.root;
    const button = rootInstance.findByType("button");
    button.props.onClick();
    expect(button.props.children).toBe("PROCEED TO CHECKOUT");
  });
});