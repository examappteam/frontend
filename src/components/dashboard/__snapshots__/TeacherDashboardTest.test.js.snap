// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherDashboard it matches the snapshot 1`] = `
<div>
  <div
    className="pure-g"
  >
    <div
      className="pure-u-1-3"
    >
      <div
        className="padded-box"
      >
        <div
          className="pure-menu pure-menu-scrollable scrollable-list-menu"
        >
          <h3
            className="pure-menu-link pure-menu-heading"
          >
            My classes
          </h3>
          <ul
            className="pure-menu-list"
          >
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Basics of Java
                    <a
                      href="/course_view"
                      onClick={[Function]}
                    >
                      <i
                        className="fa fa-arrow-circle-right fa-lg list-button"
                      />
                    </a>
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Basic beginner course
                    
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Object oriented analysis
                    
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Advanced maths
                    
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Best course ever
                    
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Java for experts
                    
                  </p>
                  <p>
                    
                  </p>
                </div>
              </li>
            </div>
          </ul>
        </div>
        <button
          className="pure-button pure-button-primary"
          onClick={[Function]}
        >
          Create new class
        </button>
        <div
          className="back-drop"
          style={
            Object {
              "opacity": "0",
              "transform": "translateZ(-100vh)",
              "visibility": "hidden",
            }
          }
        >
          <div
            className="modal-wrapper-md"
            style={
              Object {
                "opacity": "0",
                "transform": "translateZ(-100vh)",
                "visibility": "hidden",
              }
            }
          >
            <div
              className="modal-header-md"
            >
              <p
                className="close-container"
                onClick={[Function]}
              >
                <i
                  className="close-modal-btn fas fa-times"
                />
              </p>
            </div>
            <div
              className="modal-body-md"
            >
              <div
                className="padded-box"
              >
                <form
                  className="pure-form pure-form-aligned"
                >
                  <label>
                    Name of the class: 
                  </label>
                  <input
                    name="courseName"
                    onChange={[Function]}
                    placeholder="eg. Basic Electronics"
                    type="text"
                    value=""
                  />
                </form>
                <button
                  className="pure-button btn-continue"
                  onClick={[Function]}
                >
                  Create class
                </button>
              </div>
            </div>
            <button
              className="pure-button btn-cancel"
              onClick={[Function]}
            >
              Cancel
            </button>
          </div>
        </div>
        <button
          className="pure-button pure-button-disabled"
        >
          Delete selected
        </button>
      </div>
    </div>
    <div
      className="pure-u-1-3"
    >
      <div
        className="padded-box"
      >
        <div
          className="pure-menu pure-menu-scrollable scrollable-list-menu"
        >
          <h3
            className="pure-menu-link pure-menu-heading"
          >
            Ready for evaluation
          </h3>
          <ul
            className="pure-menu-list"
          >
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Programming test
                    
                  </p>
                  <p>
                    Basics of Java
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Beginner exam
                    
                  </p>
                  <p>
                    Basic beginner course
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Analysis pt. 1
                    
                  </p>
                  <p>
                    Object oriented analysis
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Vector calculation
                    
                  </p>
                  <p>
                    Advanced maths
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Best exam ever
                    
                  </p>
                  <p>
                    Best course ever
                  </p>
                </div>
              </li>
            </div>
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Live coding exam
                    
                  </p>
                  <p>
                    Java for experts
                  </p>
                </div>
              </li>
            </div>
          </ul>
        </div>
        <a
          href="/exam_grading"
          onClick={[Function]}
        >
          <button
            className="pure-button pure-button-primary"
          >
            Evaluate selected
          </button>
        </a>
      </div>
    </div>
    <div
      className="pure-u-1-3"
    >
      <div
        className="padded-box"
      >
        <div
          className="pure-menu pure-menu-scrollable scrollable-list-menu"
        >
          <h3
            className="pure-menu-link pure-menu-heading"
          >
            My own exams
          </h3>
          <ul
            className="pure-menu-list"
          >
            <div
              className="listMenuItem"
              onClick={[Function]}
            >
              <li
                className="pure-menu-item"
              >
                <div
                  className="pure-menu-link"
                >
                  <p>
                    Loading...
                    
                  </p>
                  <p />
                </div>
              </li>
            </div>
          </ul>
        </div>
        <a
          href="/create_exam"
          onClick={[Function]}
        >
          <button
            className="pure-button pure-button-primary"
          >
            Create new exam
          </button>
        </a>
        <button
          className="pure-button pure-button-disabled"
          id="delete-exam-disabled-button"
        >
          Delete selected
        </button>
      </div>
    </div>
  </div>
</div>
`;
